python/cudf/cudf/tests/test_dataframe.py

```diff
@@ -9068,6 +9068,77 @@ def test_dataframe_add_suffix():
     assert_eq(got, expected)
 
 
+@pytest.mark.parametrize(
+    "data",
+    [
+        np.random.RandomState(seed=10).randint(-50, 50, (25, 30)),
+        np.random.RandomState(seed=10).random_sample((4, 4)),
+        np.array([1.123, 2.343, 5.890, 0.0]),
+        [True, False, True, False, False],
+        {"a": [1.123, 2.343, np.nan, np.nan], "b": [None, 3, 9.08, None]},
+    ],
+)
+@pytest.mark.parametrize("periods", range(-10, 10))
```

Comment:
Would it be sufficient to test fewer values? Running 20 different values seems excessive.
```suggestion
@pytest.mark.parametrize("periods", (-5, -1, 0, 1, 5))
```